### ISSUE TITLE
Undocumented change of RectangleComponent default color in Flame 1.5.0/1.6.0 

### ISSUE THREAD

**Issue by tksoh** [flame-engine/flame#2347]
# Current bug behavior

In Flame 1.5.0/1.6.0, RectangleComponent has default color of Colors.white.

# Expected behavior

In flame 1.4.0, RectangleComponent has default color of Colors.transparent. I have to call setColor(Colors.transparent) to recreate Flame 1.4.0 behavior as workaround in my project.

I'm not sure if this breaking change is intentional, as I wasn't able to find the change in CHANGELOG.md.

# Steps to reproduce

None required. Potentially just a documentation bug.

# Flutter doctor output

Output of: flutter doctor -v
[√] Flutter (Channel stable, 3.7.3, on Microsoft Windows [Version 10.0.22621.1265], locale en-US)
    • Flutter version 3.7.3 on channel stable at C:\src\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 9944297138 (7 days ago), 2023-02-08 15:46:04 -0800
    • Engine revision 248290d6d5
    • Dart version 2.19.2
    • DevTools version 2.20.1

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at C:\Users\teeka\AppData\Local\Android\sdk
    • Platform android-33, build-tools 33.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Build Tools 2019 16.11.16)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools
    • Visual Studio Build Tools 2019 version 16.11.32602.291
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2021.2)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)

[√] VS Code, 64-bit edition (version 1.75.1)
    • VS Code at C:\Program Files\Microsoft VS Code
    • Flutter extension version 3.58.0

[√] Connected device (4 available)
    • RMX2001 (mobile)  • 192.168.0.104:5555 • android-arm64  • Android 11 (API 30)
    • Windows (desktop) • windows            • windows-x64    • Microsoft Windows [Version 10.0.22621.1265]
    • Chrome (web)      • chrome             • web-javascript • Google Chrome 109.0.5414.121
    • Edge (web)        • edge               • web-javascript • Microsoft Edge 109.0.1518.55

[√] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!

# More environment information

* Flame version: 1.6.0
* Platform affected: android (not checked on other platforms)
* Platform version affected: android 11

# Log information

N/A

# More information

N/A

**Comment by spydon** [comment]
From looking at the code from v1.4.0 (and earlier) it looks like we have always drawn the shapes in white since it uses the `HasPaint` mixin which has always defaulted to white.
The rectangle is rendered [here](https://github.com/flame-engine/flame/blob/v1.4.0/packages/flame/lib/src/geometry/polygon_component.dart#L179) and it uses the paint variable from [here](https://github.com/flame-engine/flame/blob/v1.4.0/packages/flame/lib/src/components/mixins/has_paint.dart#L15), these links go to the v1.4.0 version.

So most likely there is a change on your side that has resulted in the different color, if you can create a minimal reproducible example for v1.4.0 I can have a closer look. :)

**Comment by tksoh** [comment]
Thanks for the quick response. I dived further into my code, and noticed something strange, though I'm not sure what really is going on due to my limited knowledge with Flame. 

I created an example in this [repo](https://github.com/tksoh/flame_rectangle_color) to show the difference between flame 1.6.0 (branch flame-1.6.0) and 1.4.0 (branch flame-1.4.0), which you can see from the videos below (captured on Windows 11). The red dots are for checking to make sure the size of the Shield component indeed cover the whole app screen.

https://user-images.githubusercontent.com/494222/219542096-32318928-b947-44f5-b18b-71df2c3351ca.mp4

https://user-images.githubusercontent.com/494222/219542297-b49a1079-4d81-4a22-88e3-e6adff2468d8.mp4

**Comment by ufrshubham** [comment]
@tksoh the repo that you linked doesn't have any code apart from a default FlameGame.

**Comment by tksoh** [comment]
@ufrshubham did you look at the two branches I mentioned?

**Comment by tksoh** [comment]
@ufrshubham these are the branches:

https://github.com/tksoh/flame_rectangle_color/tree/flame-1.4.0
https://github.com/tksoh/flame_rectangle_color/tree/flame-1.6.0

**Comment by ufrshubham** [comment]
@tksoh Sorry, I completely missed those branches earlier 😅. Just verified those branches and I am pretty sure the behavior you are seeing is because of this fix #2167. Basically, before that fix, the `RectangleComponent` was not updating it's vertices if its size was modified after construction. To achieve the results of 1.6 in 1.4, you would have needed to set the `size` while construction. In your case, if you modify the `Shield` constructor to:
```dart
Shield({super.size});
```
and then set it while adding it to the game:
```dart
add(Shield(size: size));
```
you'll notice that you get the correct output in 1.4

In short, `RectangleComponent` was always rendering in white color by default but just the vertex recalculations were out of sync on size change. Also, this change was logged in [1.5.0](https://pub.dev/packages/flame/changelog#150)'s changelog.

**Comment by spydon** [comment]
Aah good catch @ufrshubham, forgot about that bug fix!

**Comment by tksoh** [comment]
@ufrshubham thanks for helping to clarify this. This might also help explain some of the confusion I had with regard to some strange behavior I observed earlier with RectangleComponent.

BTW, just thinking out load...  while the changelog did mention about #2167, it doesn't really tell us what that translate into the actual impact it might have on the project that were unknowingly build around the bug, and the title of that PR doesn't really give out a lot of info to signal something is going to break in our projects. Perhaps this is some area that can be improved upon. Granted, documentation is never easy.

In any case, I'm closing this issue. Thanks again. And keep up with the great work!

**Comment by ufrshubham** [comment]
> it doesn't really tell us what that translate into the actual impact it might have on the project that were unknowingly build around the bug

Yeah, it is always tricky with such fixes. For this one specifically, I never thought anyone would depend on the wrong output 😅. I guess you never know how someone is using a feature until it is modified.

> the title of that PR doesn't really give out a lot of info to signal something is going to break in our projects

Well technically that PR was to fix something that wasn't working. So we didn't bother marking it as a breaking change.

**Comment by tksoh** [comment]
BTW, I can't find the info w.r.t to the white default color for the components in the documentation. Are they hiding somehow? If so, shouldn't they deserve a mention too?

**Comment by spydon** [comment]
> BTW, I can't find the info w.r.t to the white default color for the components in the documentation. Are they hiding somehow? If not, shouldn't they deserve a mention too?
> 

Yeah, it would probably be good to mention that on the components page and mention the `HasPaint` mixin too.
